### ISSUE TITLE
[bug] Apply void operation disregards object's flipped state

### ISSUE THREAD

**Issue by Vanuan** [IfcOpenShell/IfcOpenShell#4950]
When applying a void of a flipped object (particularly doors and windows) to a wall, the operation does not consider the flipped state of the object. This results in the object reverting to its original, non-flipped orientation after the void is applied.

**Steps to Reproduce:**
1. Create a wall.
2. Add a door to the wall.
3. Select only the door.
4. Flip the door using the Shift+F shortcut.
5. Select both the door and the wall.
6. Apply void (Shift+O).
7. Observe that the door has reverted to its original, non-flipped state.

**Expected Behavior:**
The apply void operation should maintain the door's current orientation, including its flipped state.

**Actual Behavior:**
The apply void operation resets the door to its original orientation, disregarding its flipped state.

**Additional Information:**
- The problem is related to the apply void functionality not considering the flipped state of objects.
- This is separate from #4949

**Comment by zhukoven** [comment]
I've noticed the same bug with voids in beams and columns. So it's probably a generic issue with voids in the current build.

**Comment by theoryshaw** [comment]
might be related: https://github.com/IfcOpenShell/IfcOpenShell/issues/3254

**Comment by BIMvoice** [comment]
I could not reproduce this on the current v0.8.0. Following the exact steps (wall → door → flip the door with Shift+F → select door+wall → Apply Void with Shift+O), the door keeps its flipped orientation after the void is applied. I verified with plain ifcopenshell that the door's placement stays flipped (180° about Z), and the same holds for a flipped beam/column that is then voided.

This looks to have been resolved by intervening work: the fill placement is now re-derived from the wall face the fill sits on, and #3254 (which was linked here) is closed. Could you confirm whether you still see it on a recent build, and if so attach the .ifc and your Bonsai/Blender versions? Otherwise I think this can be closed.

**Comment by Vanuan** [comment]
Did you test it manually? Could you provide a screenshot?

**Comment by steverugi** [comment]
@Vanuan 
see clip, no issue when flipping the door, no need to use shift+O either since the opening is created when the door is added to the wall
hope it helps
cheers

https://github.com/user-attachments/assets/4c8eac8e-df59-4460-86d7-6992e767a8ab

**Comment by BIMvoice** [comment]
Agreed with @steverugi, and the clip shows it clearly. That matches exactly what I found.

I also reproduced your exact steps on current v0.8.0, wall, door, flip with Shift+F, select both, Apply Void with Shift+O, and the door kept its flipped orientation, it did not revert.

And to @steverugi's point, which is really the crux: Bonsai already creates the opening automatically when you add the door to the wall, so Shift+O is not needed for a normal door at all. Could you explain why you apply the void by hand in your case? Is the door created away from the wall first, or is it not auto voiding in your setup, or is there something specific about your scene? That probably decides whether there is still a real bug and where.

Happy to share a screenshot of my test and dig into your actual scenario if you can tell me the workflow that leads you to apply void by hand.

**Comment by Vanuan** [comment]
As I remember, back in the day, if you modified the filing dimensions you had to manually apply void to recalculate the opening.

**Comment by Vanuan** [comment]
Yeah, this was part of my documentation effort: https://github.com/IfcOpenShell/IfcOpenShell/blame/9ae79b42dd27c60db4c86bede45ce52957225dbc/src/bonsai/docs/guides/authoring/basic_modeling/door.rst#L111

If it's no longer required, feel free to edit the doc.

**Comment by steverugi** [comment]
@Vanuan I'm closing this Issue then, thanks
